### PR TITLE
implement crud-batch-cycle

### PR DESCRIPTION
**Pull request by dhardison85**
implement `crud-batch-cycle`

### CODE REVIEW

File: domain/batch/entity.go [hunk 1]
```diff
@@ -34,3 +34,17 @@ type Pool struct {
...
+type BatchCycle struct {
+	ID      uuid.UUID   `json:"id"`
+	Batch   Batch       `json:"batch"`
+	BatchID uuid.UUID   `json:"-"`
+	Pool    Pool        `json:"pool"`
+	PoolID  uuid.UUID   `json:"-"`
+	Weight  float64     `json:"weight"`
+	Amount  float64     `json:"amount"`
+	Start   time.Time   `json:"start"`
+	Finish  null.String `json:"finish"`
```
Comment (ncrypthic) [inline domain/batch/entity.go]: Do you mean `null.Time`?

File: handler/handler.go [hunk 2]
```diff
@@ -87,6 +87,8 @@ func (h *BatchHandler) StoreGrowthBatch(c *gin.Context) {
 		//compare uuid to batch
 		//save if valid
 		var uid, err = uuid.FromString(id)
+		log.Print(uid, "\n")
+		log.Print(batch.ID)
```
Comment (ncrypthic) [inline handler/handler.go]: Unnecessary?

File: handler/handler.go [hunk 3]
```diff
@@ -275,6 +275,80 @@ func (h *BatchHandler) RemoveGrowthPoolByIDs(c *gin.Context) {
...
+	}
+	return
+}
+
+func (h *BatchHandler) StoreGrowthBatchCycle(c *gin.Context) {
+
+	var id = c.Params.ByName("id")
+	var bc batch.BatchCycle
+	c.BindJSON(&bc)
+	log.Print("batch:", bc.Batch)
```
Comment (ncrypthic) [inline handler/handler.go]: Unnecessary?

File: main.go [hunk 4]
```diff
@@ -63,6 +63,11 @@ func main() {
 		growth.PUT("/pool/:id", batchHandler.StoreGrowthPool)
 		growth.DELETE("/pool", batchHandler.RemoveGrowthPoolByIDs)
 		growth.DELETE("/pool/:id", batchHandler.RemoveGrowthPoolByID)
+		//batch cycle
+		growth.GET("/batch-cycle", batchHandler.ResolveGrowthBatchCyclePage)
```
Comment (ncrypthic) [inline main.go]: Can we make the url more declarative? E.g `/batches/:batchId/cycles/:cycleId` :thinking:

Comment (dhardison85) [reply main.go]: but we don't have a cycle_id, we only have batch_cycle_id (which is primary key), batch_id and pool_id, please refer to database design.. :cry:

Comment (ncrypthic) [reply main.go]: Then `/batches/:batchId/cycles/:batchCycleId

**Pull request merged**